rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java

```diff
@@ -0,0 +1,209 @@
+     * bounds
+     */
+    @Override
+    public LockWhitelistEntry getLockWhitelistEntryByIndex(int whitelistIndex) {
+        List<LockWhitelistEntry> whitelistEntries = whitelistStorageProvider.getLockWhitelist().getAll();
+
+        if (whitelistIndex < 0 || whitelistIndex >= whitelistEntries.size()) {
+            return null;
+        }
+        return whitelistEntries.get(whitelistIndex);
+    }
+
+    @Override
+    public LockWhitelistEntry getLockWhitelistEntryByAddress(String addressBase58) {
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+
+            return whitelistStorageProvider.getLockWhitelist().get(address);
+        } catch (AddressFormatException e) {
+            logger.warn(INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return null;
+        }
+    }
+
+    /**
+     * Adds the given address to the lock whitelist.
+     *
+     * @param addressBase58    the base58-encoded address to add to the whitelist
+     * @param maxTransferValue the max amount of satoshis enabled to transfer for this address
+     * @return 1 SUCCESS, -1 ADDRESS_ALREADY_WHITELISTED, -2 INVALID_ADDRESS_FORMAT,
+     * -10 GENERIC_ERROR, 0 UNKNOWN_ERROR.
+     */
+    @Override
+    public int addOneOffLockWhitelistAddress(Transaction tx, String addressBase58, BigInteger maxTransferValue) {
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+            Coin maxTransferValueCoin = Coin.valueOf(maxTransferValue.longValueExact());
+            LockWhitelistEntry whiteListEntry = new OneOffWhiteListEntry(address, maxTransferValueCoin);
+
+            return this.addLockWhitelistAddress(tx, whiteListEntry);
+        } catch (AddressFormatException e) {
+            logger.warn(INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return INVALID_ADDRESS_FORMAT.getCode();
+        }
+    }
+
+    @Override
+    public int addUnlimitedLockWhitelistAddress(Transaction tx, String addressBase58) {
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+            LockWhitelistEntry whiteListEntry = new UnlimitedWhiteListEntry(address);
+
+            return this.addLockWhitelistAddress(tx, whiteListEntry);
+        } catch (AddressFormatException e) {
+            logger.warn(INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return INVALID_ADDRESS_FORMAT.getCode();
+        }
+    }
+
+    /**
+     * Removes the given address from the lock whitelist.
+     *
+     * @param addressBase58 the base58-encoded address to remove from the whitelist
+     * @return 1 SUCCESS, -1 ADDRESS_NOT_EXIST, -2 INVALID_ADDRESS_FORMAT, -10 GENERIC_ERROR,
+     * 0 UNKNOWN_ERROR.
+     */
+    @Override
+    public int removeLockWhitelistAddress(Transaction tx, String addressBase58) {
+        if (isAuthorizedLockWhitelistChange(tx)) {
```

Comment:
you [removed](https://github.com/rsksmart/rskj/pull/2460/files#diff-61e5c53e59bedc5790b677df9fb9db0fbe260d86030c6caa00fb9b12a6afc685L2135) the ! condition.

Reply:
Careful here @wilmerrootstock , why did you change this? This is consensus code, should just copy the existing logic from BridgeSupport to this class

Reply:
Hey, guys! @marcos-iov @julia-zack 
Yes, I removed the "!" expression. It was suggested by IntelliJ to make more readable the code. I changed the name from isAuthorizedLockWhitelistChange to isNotAuthorizedLockWhitelistChange and inverted the internal condition. Please, let me know if this makes sense or if I should revert this change.

Reply:
you are throwing a GENERIC_ERROR when `isAuthorizedLockWhitelistChange`. That's wrong.

The code that was there wasn't _is**Not**AuthorizedLockWhitelistChange_, but _isAuthorizedLockWhitelistChange_, and it was checking for it to be false. So you reversed the logic. See the link I referred above.

Reply:
Done! I created a commit with this change only, to isolate it from other changes to take more control of it:

- Reverted improvement since at this stage it is not needed. The improvement was made based on avoiding negation statements into conditionals to make the code more readable.
- Refactor reverted to leave it like it was before refactoring.